### ISSUE TITLE
CA1416 with guard attributes

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/roslyn-analyzers#7647]
### Analyzer

**Diagnostic ID**: warning CA1416: This call site is reachable on: 'macOS/OSX', 'tvos' 12.2 and later. 'TestType.GpuRegistryId' is only supported on: 'tvos' 13.0 and later.

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

```
$ .dotnet/dotnet --version
10.0.100-preview.3.25167.3
```
### Describe the bug

Incorrect CA1416 when trying to access API in code block protected using guard attributes.

### Steps To Reproduce

Run this unit test: https://github.com/rolfbjarne/roslyn-analyzers/commit/3c8fed4303a856c2e81303f576c3ce39c3238dd6

### Expected behavior

Unit test should pass

### Actual behavior

Unit test fails:

> Diagnostics:
// /0/Test0.cs(13,32): warning CA1416: This call site is reachable on: 'macOS/OSX', 'tvos' 12.2 and later. 'TestType.GpuRegistryId' is only supported on: 'tvos' 13.0 and later.

## Additional context

This is not fixed by #7569 nor #7644.

**Comment by rolfbjarne** [comment]
CC @buyaa-n

**Comment by buyaa-n** [comment]
> This is not fixed by https://github.com/dotnet/roslyn-analyzers/pull/7569 nor https://github.com/dotnet/roslyn-analyzers/pull/7644.

This is related to #7569

It is kind of an annotation issue, in the repro you shared `GpuRegistryId` attribute support list will have `"tvos13.0" and "macos12.0"`. The `[SupportedOSPlatform ("macos")]` attribute on the property will be ignored because assembly has `[SupportedOSPlatform ("macos12.0")]` attribute and child (nested) API could not widen that support.

On the other side the Guard attributes will process `[SupportedPlatformGuard ("macos")]` and `[SupportedOSPlatformGuard ("tvos13.0")]` as is, because flow analysis doesn't evaluate parent (assembly) attributes.

That means the `if (IsAtLeastXcode11) { }` block is reachable on `macos0.0` and `tvos13.0` but the referenced API `GpuRegistryId` is only supported on `"tvos13.0" and "macos12.0"`. If the guard had `[SupportedOSPlatformGuard ("macos12.0")]` or above instead of `[SupportedOSPlatformGuard ("macos")]` there should have been no issue

```cs
using System;
using System.Runtime.Versioning;

// removing the assembly: attributes make the test work
[assembly: SupportedOSPlatform ("tvos12.2")]
[assembly: SupportedOSPlatform ("macos12.0")]

partial class TestType {
    void DoSomething ()
    {
        if (IsAtLeastXcode11) {
            Console.WriteLine (GpuRegistryId); // warns here because guard allows `macos0.0` but API is supported from `macos12.0`
        }
    }

    [SupportedOSPlatform ("macos")]
    [SupportedOSPlatform ("tvos13.0")]
    public ulong? GpuRegistryId { get; private set; }

    [SupportedOSPlatformGuard ("macos")]
    [SupportedOSPlatformGuard ("tvos13.0")]
    internal static bool IsAtLeastXcode11 {
        get {
            return true;
        }
    }
}
```
Not sure if we want to tweak the analyzer for this or ask the user to add a proper guard, i.e `[SupportedOSPlatformGuard ("macos12.0")]`

**Comment by rolfbjarne** [comment]
> ask the user to add a proper guard, i.e `[SupportedOSPlatformGuard ("macos12.0")]`

I can do that, thanks for the analysis!
